ExampleMod/Content/Projectiles/Minions/ExampleSentry.cs

```diff
@@ -0,0 +1,173 @@
+			Projectile.height = 30;
+			Projectile.DamageType = DamageClass.Summon;
+			Projectile.sentry = true; // Sets the weapon as a sentry for sentry accessories to properly work.
+			Projectile.timeLeft = Projectile.SentryLifeTime; // Sentries last 10 minutes
+			Projectile.ignoreWater = true;
+			Projectile.netImportant = true; // Sentries need this so they are synced to newly joining players
+
+			// The texture is 54 pixels wide, but we set width to 46 and DrawOffsetX to -6 so it doesn't look weird hanging off the edge of tiles (because it is oval shaped).
+			DrawOffsetX = -6;
+		}
+
+		public override bool TileCollideStyle(ref int width, ref int height, ref bool fallThrough, ref Vector2 hitboxCenterFrac) {
+			fallThrough = false; // Allow this projectile to collide with platforms
+			return true;
+		}
+
+		public override bool OnTileCollide(Vector2 oldVelocity) {
+			return false; // Prevent tile collision from killing the projectile
+		}
+
+		public override Color? GetAlpha(Color lightColor) {
+			// Always draw fully bright. This is important because sentries can usually be placed inside tiles where it would be dark.
+			return Color.White;
+		}
+
+		// This AI will function as a static sentry, and will not move. If you would like to know how to do more advanced minion AI, check out ExampleSimpleMinion.cs.
+		public override void AI() {
+			const int ShootFrequency = 60; // How long the sentry waits between shots.
+			const int TargetingRange = 50 * 16; // The sentry's targeting range, 50 tiles.
+			const float FireVelocity = 10f; // The velocity the sentry's shot projectile will travel.
+
+			// Code to run when spawned
+			if (JustSpawned) {
+				JustSpawned = false;
+				ShootTimer = ShootFrequency * 1.5f; // Delay the first shot slightly
+
+				// The sound that Frost Hydra, Spider Turret, and Houndius Shootius play when spawned. Optional.
+				SoundEngine.PlaySound(SoundID.Item46, Projectile.position);
+
+				// Dust indicating the sentry spawned. Optional.
+				for (int i = 0; i < 50; i++) {
+					Vector2 speed = Main.rand.NextVector2CircularEdge(1f, 1f);
+					Dust d = Dust.NewDustPerfect(Projectile.Center, DustID.BlueCrystalShard, speed * 4, Scale: 1.5f);
+					d.noGravity = true;
+				}
+			}
+
+			// Spawn dust randomly
+			if (Main.rand.NextBool(10)) {
+				Dust dust = Dust.NewDustDirect(Projectile.position + Projectile.velocity, Projectile.width, Projectile.height, DustID.Firework_Blue, Projectile.oldVelocity.X * 0.5f, Projectile.oldVelocity.Y * 0.5f);
+				dust.noGravity = true;
+				dust.velocity *= 0.8f;
+			}
+
+			// Gravity
+			Projectile.velocity.X = 0f;
+			if (!Floating) { // Only apply gravity if canPlaceInAir from ExampleSentryItem.Shoot was false
+				Projectile.velocity.Y += 0.2f;
+				if (Projectile.velocity.Y > 16f) {
+					Projectile.velocity.Y = 16f;
+				}
+			}
+
+			// Find an enemy to target.
+			bool hasValidTarget = false;
+			float closestTargetDistance = TargetingRange;
+			TargetNPCIndex = -1;
```

Comment:
Couldn't `TargetNPCIndex` be a local variable? If so, would it be better to just make it `NPC targetNPC`?